e2e/src/pages/navigate.po.ts

```diff
@@ -1,20 +1,28 @@
-import { by, element } from 'protractor';
-import { AppPage } from './app.po';
+import {by, element, ElementFinder} from 'protractor';
+import {AppPage} from './app.po';
 
-export class NavigatePage extends AppPage{
+export class NavigatePage extends AppPage {
+  public pageNumber: ElementFinder;
+  public next: ElementFinder;
+  public previous: ElementFinder;
 
-  async goToNextPage() {
-    await this.waitForElement(by.id('next'));
-    await this.clickElement(by.id('next'));
-  }
-
-  async goToPreviousPage() {
-    await this.waitForElement(by.id('previous'));
-    await this.clickElement(by.id('previous'));
+  constructor() {
+    super();
+    this.next = element(by.id('next'));
```

Comment:
this.next is set to a promise. when you do goToNextPage(), you're going to be clicking on a promise, without knowing if it's resolved or not. That means it will sometimes work, and sometimes not work.
to fix, you could set
this.next = await element(by.id('next'));
and do something similar for all other elements that you're defining

Reply:
Thanks @JeroenRijks  I will incorporate your review comments

Reply:
I cannot declare constructor() as async as the compiler throwing an error message